src/peer_digest.cc

```diff
@@ -491,7 +487,7 @@ peerDigestFetchReply(void *data, char *buf, ssize_t size)
 
             // stay with the old in-memory digest
             peerDigestFetchStop(fetch, buf, "Not modified");
-            fetch->state = DIGEST_READ_DONE;
```

Comment:
peerDigestFetchStop() indirectly deletes `fetch`. We will post a followup cleanup PR (that was [promised](https://github.com/squid-cache/squid/pull/1857#discussion_r1672250016) in the previous PR in this series) that will make that fact obvious in this and similar callers, so please do not ask to add `// deletes fetch` C++ comments or similar adjustments in this bug-fixing PR.

P.S. This comment does not request any code changes.